write.go

```diff
@@ -38,3 +85,394 @@ func toOptSet(opts ...WriteOption) *writeOptSet {
 	}
 	return optSet
 }
+
+func writeFileHeader(w dicomio.Writer, ds *Dataset, metaElems []*Element, opts ...WriteOption) error {
+	// File headers are alwyas written in littleEndian explicit
+	w.SetTransferSynax(binary.LittleEndian, false)
+
+	metaBytes := &bytes.Buffer{}
+	subWriter := dicomio.NewWriter(metaBytes, binary.LittleEndian, false)
+	tagsUsed := make(map[tag.Tag]bool)
+	tagsUsed[tag.FileMetaInformationGroupLength] = true
+
+	// TODO make better structure for error checking so it's no so many lines
```

Comment:
nit: `TODO:` (with a colon), but we can cleanup in future PR if needed. In the future it'll be good to associated a bug number too e.g. `TODO(#42):` or a username `TODO(suyashkumar):`